CMocka unit tests fail in Docker due to unexpected error codes

Some CMocka unit tests are failing under specific conditions, such as when executed in Docker environments. For example:

```
[ RUN      ] test_set_auditd_config_audit_plugin_not_created_fopen_error
[  ERROR   ] --- "(1103): Could not open file 'etc/af_wazuh.conf' due to [(95)-(Operation not supported)]." != "(1103): Could not open file 'etc/af_wazuh.conf' due to [(0)-(Success)]."
unit_tests/wrappers/wazuh/shared/debug_op_wrappers.c:61: error: Check of parameter formatted_msg, function __wrap__merror failed
/src/src/unit_tests/syscheckd/whodata/test_syscheck_audit.c:677: note: Expected parameter declared here
[   LINE   ] --- unit_tests/wrappers/wazuh/shared/debug_op_wrappers.c:61: error: Failure!
[  FAILED  ] test_set_auditd_config_audit_plugin_not_created_fopen_error
[ RUN      ] test_set_auditd_config_audit_plugin_not_created_fclose_error
[  ERROR   ] --- "(1140): Could not close file 'etc/af_wazuh.conf' due to [(95)-(Operation not supported)]." != "(1140): Could not close file 'etc/af_wazuh.conf' due to [(0)-(Success)]."
unit_tests/wrappers/wazuh/shared/debug_op_wrappers.c:61: error: Check of parameter formatted_msg, function __wrap__merror failed
/src/src/unit_tests/syscheckd/whodata/test_syscheck_audit.c:729: note: Expected parameter declared here
[   LINE   ] --- unit_tests/wrappers/wazuh/shared/debug_op_wrappers.c:61: error: Failure!
[  FAILED  ] test_set_auditd_config_audit_plugin_not_created_fclose_error
```

This may be caused by uninitialized variables leading to unexpected error codes (e.g., errno 95 instead of 0). Please investigate the cause and ensure variables are properly initialized to avoid inconsistent test results, especially in containerized environments.

--- a/src/unit_tests/syscheckd/whodata/test_syscheck_audit.c
+++ b/src/unit_tests/syscheckd/whodata/test_syscheck_audit.c
@@ -676,6 +676,8 @@ void test_set_auditd_config_audit_plugin_not_created_fopen_error(void **state) {
 
     expect_string(__wrap__merror, formatted_msg, "(1103): Could not open file 'etc/af_wazuh.conf' due to [(0)-(Success)].");
 
+    errno = 0;
+
     int ret;
     ret = set_auditd_config();
 
@@ -727,6 +729,8 @@ void test_set_auditd_config_audit_plugin_not_created_fclose_error(void **state) 
     will_return(__wrap_fclose, -1);
 
     expect_string(__wrap__merror, formatted_msg, "(1140): Could not close file 'etc/af_wazuh.conf' due to [(0)-(Success)].");
+
+    errno = 0;
 
     int ret;
     ret = set_auditd_config();

